src/sources/pulsar.rs

```diff
@@ -0,0 +1,624 @@
+    #[configurable(metadata(docs::examples = "pulsar://127.0.0.1:6650"))]
+    #[serde(alias = "address")]
+    endpoint: String,
+
+    /// The Pulsar topic names to read events from.
+    #[configurable(metadata(docs::examples = "[persistent://public/default/my-topic]"))]
+    topics: Vec<String>,
+
+    /// The Pulsar consumer name.
+    #[configurable(metadata(docs::examples = "consumer-name"))]
+    consumer_name: Option<String>,
+
+    /// The Pulsar subscription name.
+    #[configurable(metadata(docs::examples = "subscription_name"))]
+    subscription_name: Option<String>,
+
+    /// Priority level for a consumer to which a broker gives more priority while dispatching messages in Shared subscription type.
+    ///
+    /// The broker follows descending priorities. For example, 0=max-priority, 1, 2,...
+    ///
+    /// In Shared subscription type, the broker first dispatches messages to the max priority level consumers if they have permits. Otherwise, the broker considers next priority level consumers.
+    priority_level: Option<i32>,
+
+    /// Max count of messages in a batch.
+    batch_size: Option<u32>,
+
+    #[configurable(derived)]
+    auth: Option<AuthConfig>,
+
+    #[configurable(derived)]
+    dead_letter_queue_policy: Option<DeadLetterQueuePolicy>,
+
+    #[configurable(derived)]
+    #[serde(default = "default_framing_message_based")]
+    #[derivative(Default(value = "default_framing_message_based()"))]
+    framing: FramingConfig,
+
+    #[configurable(derived)]
+    #[serde(default = "default_decoding")]
+    #[derivative(Default(value = "default_decoding()"))]
+    decoding: DeserializerConfig,
+
+    #[configurable(derived)]
+    #[serde(default, deserialize_with = "bool_or_struct")]
+    acknowledgements: SourceAcknowledgementsConfig,
+
+    /// The namespace to use for logs. This overrides the global setting.
+    #[configurable(metadata(docs::hidden))]
+    #[serde(default)]
+    log_namespace: Option<bool>,
+}
+
+/// Authentication configuration.
+#[configurable_component]
+#[derive(Clone, Debug)]
+struct AuthConfig {
+    /// Basic authentication name/username.
+    ///
+    /// This can be used either for basic authentication (username/password) or JWT authentication.
+    /// When used for JWT, the value should be `token`.
+    #[configurable(metadata(docs::examples = "${PULSAR_NAME}"))]
+    #[configurable(metadata(docs::examples = "name123"))]
+    name: Option<String>,
+
+    /// Basic authentication password/token.
+    ///
+    /// This can be used either for basic authentication (username/password) or JWT authentication.
+    /// When used for JWT, the value should be the signed JWT, in the compact representation.
+    #[configurable(metadata(docs::examples = "${PULSAR_TOKEN}"))]
+    #[configurable(metadata(docs::examples = "123456789"))]
+    token: Option<SensitiveString>,
+
+    #[configurable(derived)]
+    oauth2: Option<OAuth2Config>,
+}
```

Comment:
Given that the user can either specify `name` and `token` _or_ `oauth2`, can we change this to an enum to make that explicit in the data model and resulting documentation?